python/ray/tests/test_basic_3.py

```diff
@@ -284,8 +284,9 @@ def get_num_workers():
         return None
 
     # Wait for "debug_state.txt" to be updated to reflect the started worker.
+    timeout_limit = 40 if sys.platform == "win32" else 10
```

Comment:
In some cases (other than those the time taken is 23ish seconds), this test takes 30ish seconds to get updated data from `debug_state.txt`. So, for safety I have set the limit to 40s.